### ISSUE TITLE
Use shell to pip to extract hashes

### ISSUE THREAD

**Issue by peterbe** [peterbe/hashin#58]
We currently rely on [`_hash_of_file`](https://github.com/peterbe/hashin/blob/77e2d93f40f7b48ea147f5b032e517cada70b96c/hashin.py#L366) to extract the checksum of a package using pip. Yuck! We should use a shell and call like this:
```python
subprocess.call(['python', '-m', 'pip', 'hash', ...])
```

Thanks @di for the tip!

**Comment by di** [comment]
I actually have something even better in the works, hang tight...

**Comment by di** [comment]
Hmm, seems like this won't fly with 2.6 for some reason:

```
$ python2.6 -m pip
/Users/di/.pyenv/versions/2.6.9/bin/python2.6: pip is a package and cannot be directly executed
```

works fine for other things though:
```
$ python2.6 -m unittest

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
```

@peterbe How do you feel about officially dropping Python 2.6 support? It's nearly 5 years past end-of-life at this point. Pip itself will be dropping support for Python 2.6 with the 10.0 release as well...

**Comment by di** [comment]
Heh, looks like you already did: https://github.com/peterbe/hashin/commit/e655e6446dc2a73afef31aea3722f583325d5ec8

**Comment by peterbe** [comment]
Thing is, about https://github.com/peterbe/hashin/commit/e655e6446dc2a73afef31aea3722f583325d5ec8, is that it only dropped 2.6 in terms of testing in Travis. The reason was that 2.6 couldn't be installed. Neither could 3.3. 

I did rearrange things in the travis and tox file. Perhaps I rushed it. [So I'll try again](https://github.com/peterbe/hashin/pull/60) Seems that you *can* get 2.6 on Travis but now it's failing on something else related to setuptools and I just don't know if I care. 

@di your `python2.6 -m unittest` ran 0 tests :)

Would it work with your 2.6 if it has pip installed separately? I.e. `download pip; cd downloaded-pip; /Users/di/.pyenv/versions/2.6.9/bin/python2.6 setup.py install`
Does  `python2.6 -m pip` work then?
